src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.MultipleWatchers.cs

```diff
@@ -0,0 +1,309 @@
+                    AutoResetEvent autoResetEvent2 = WatchCreated(watcher2, new[] { fileName }).EventOccured;
+                    AutoResetEvent autoResetEvent3 = WatchCreated(watcher3, new[] { fileName }).EventOccured;
+
+                    watcher1.EnableRaisingEvents = true;
+                    watcher2.EnableRaisingEvents = true;
+                    watcher3.EnableRaisingEvents = true;
+
+                    File.Create(fileName).Dispose();
+                    Assert.True(WaitHandle.WaitAll(new[] { autoResetEvent1, autoResetEvent2, autoResetEvent3 }, WaitForExpectedEventTimeout_NoRetry));
+
+                    File.Delete(fileName);
+                    watcher1.EnableRaisingEvents = false;
+
+                    File.Create(fileName).Dispose();
+                    Assert.False(autoResetEvent1.WaitOne(WaitForUnexpectedEventTimeout));
+                    Assert.True(WaitHandle.WaitAll(new[] { autoResetEvent2, autoResetEvent3 }, WaitForExpectedEventTimeout_NoRetry));
+                }
+            });
+        }
+
+        [OuterLoop]
+        [Fact]
+        public void FileSystemWatcher_File_Create_WatchOwnPath()
+        {
+            using (var dir = new TempDirectory(GetTestFilePath()))
+            using (var dir1 = new TempDirectory(Path.Combine(dir.Path, "dir1")))
+            using (var dir2 = new TempDirectory(Path.Combine(dir.Path, "dir2")))
+            using (var watcher1 = new FileSystemWatcher(dir1.Path, "*"))
+            using (var watcher2 = new FileSystemWatcher(dir2.Path, "*"))
+            {
+                string fileName1 = Path.Combine(dir1.Path, "file");
+                string fileName2 = Path.Combine(dir2.Path, "file");
+
+                AutoResetEvent autoResetEvent1 = WatchCreated(watcher1, new[] { fileName1 }).EventOccured;
+                AutoResetEvent autoResetEvent2 = WatchCreated(watcher2, new[] { fileName2 }).EventOccured;
+
+                watcher1.EnableRaisingEvents = true;
+                watcher2.EnableRaisingEvents = true;
+
+                File.Create(fileName1).Dispose();
+                Assert.True(autoResetEvent1.WaitOne(WaitForExpectedEventTimeout_NoRetry));
+                Assert.False(autoResetEvent2.WaitOne(WaitForUnexpectedEventTimeout));
+
+                File.Create(fileName2).Dispose();
+                Assert.True(autoResetEvent2.WaitOne(WaitForExpectedEventTimeout_NoRetry));
+                Assert.False(autoResetEvent1.WaitOne(WaitForUnexpectedEventTimeout));                
+            }
+        }
+
+        [OuterLoop]
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void FileSystemWatcher_File_Create_ForceLoopRestart(bool useExistingWatchers)
+        {
+            ExecuteWithRetry(() =>
+            {
+                string fileName = Path.Combine(TestDirectory, "file");
+                AutoResetEvent[] autoResetEvents = new AutoResetEvent[64];
+                FileSystemWatcher[] watchers = new FileSystemWatcher[64];
```

Comment:
We should probably dispose all of these in a finally: that way we'll promptly tear them down when the test completes, even in the case of an exception or assert failure.

Reply:
Done